ahnlich/ai/src/error.rs

```diff
@@ -7,12 +7,38 @@ pub enum AIProxyError {
     StoreNotFound(StoreName),
     #[error("Store {0} already exists")]
     StoreAlreadyExists(StoreName),
+
     #[error("Proxy Errored with {0} ")]
     StandardError(String),
+
     #[error("Proxy Errored with {0} ")]
     DatabaseClientError(String),
     #[error("Reserved key {0} used")]
     ReservedError(String),
     #[error("Unexpected DB Response {0} ")]
     UnexpectedDBResponse(String),
+    #[error("Cannot Query Using Input. Store expects [{store_query_model_type}], but input type [{storeinput_type}] was provided")]
+    StoreQueryTypeMismatchError {
+        store_query_model_type: AIStoreInputTypes,
+        storeinput_type: AIStoreInputTypes,
+    },
+    #[error("Cannot Set Input. Store expects [{store_index_model_type}], input type [{storeinput_type}] was provided")]
+    StoreSetTypeMismatchError {
+        store_index_model_type: AIStoreInputTypes,
+        storeinput_type: AIStoreInputTypes,
+    },
+
+    #[error("Max Token Exceeded. Model Expects [{model_embedding_size}], input type was [{input_token_size}] ")]
+    TokenExceededError {
+        model_embedding_size: usize,
```

Comment:
Model embedding size has nothing to do with tokens exceeded

Model embeddings is the output of the model, while preprocessing is basically us trying to get the user input to fit the allowed input of the model